### ISSUE TITLE
SocketUtils is deprecated and will be removed in Spring Security 6

### ISSUE THREAD

**Issue by nbaars** [WebGoat/WebGoat#1298]
We should probably just copy and simplify the current Spring implementation. 
Starting on 8080/9090 has preference over supplying port 0 and pointing to users they should look at the log for the actual port binding. This introduces too much complexity and questions.

**Comment by nbaars** [comment]
https://github.com/spring-projects/spring-framework/issues/28052

**Comment by nbaars** [comment]
https://github.com/spring-projects/spring-framework/issues/28052

**Comment by aolle** [comment]
If I understood the idea is to start always on 8080/9090, if no webgoat.port or webwolf.port have been provided. If those ports are not available, in my opinion, the startup has to fail. 

I've been checking the code and it has defined a MAX_PORT variable set to 9999 which is used in combination of the SocketUtils. In my opinion, I doesn't follow the "standard practices" so to speak. I agree that it introduces complexity and IMO maybe is not to clear, easy to figure out and the user is losing the real control.

On the other side, the SocketUtils.findAvailableTcpPort method is not the same than supplying port 0. SocketUtils tests the port availability and closes the socket if it was successful; the port 0 relies on the kernel. I don't think it is a good idea for the **main** binding class of a web application (which is a different scenario than using a random port on a testing/integration phase).

In my opinion, the MAX_PORT and SocketUtils stuff should be removed.

**Comment by nbaars** [comment]
> If I understood the idea is to start always on 8080/9090, if no webgoat.port or webwolf.port have been provided. If those ports are not available, in my opinion, the startup has to fail.
Indeed, solves the issue at hand I will remove the code and update the documentation accordingly.

**Comment by nbaars** [comment]
Removed all the code, the error message is clear enough if the port is taken:

```
2022-07-19 22:36:10.196 ERROR 28952 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Web server failed to start. Port 9999 was already in use.

Action:

Identify and stop the process that's listening on port 9999 or configure this application to listen on another port.


Process finished with exit code 1
```

**Comment by aolle** [comment]
Very nice!

**Comment by nbaars** [comment]
Closing as we released 2023.3
